tracer/src/Datadog.Trace/Util/Http/HttpRequestUtils.cs

```diff
@@ -4,22 +4,25 @@
 // </copyright>
 
 using System;
-using Datadog.Trace.Configuration;
-using Datadog.Trace.Util.Http.QueryStringObfuscation;
+using Datadog.Trace.Tagging;
 
 namespace Datadog.Trace.Util.Http
 {
-    internal class HttpRequestUtils
+    internal static class HttpRequestUtils
     {
         private const string NoHostSpecified = "UNKNOWN_HOST";
 
-        internal static string GetUrl(string scheme, string host, string pathBase, string path, string queryString, QueryStringManager queryStringManager = null)
+        internal static LazyOrString GetUrl(string scheme, string host, string pathBase, string path, string queryString, QueryStringManager queryStringManager = null)
         {
-            if (queryStringManager != null)
+            if (queryStringManager?.ShouldObfuscate(queryString) ?? false)
```

Comment:
I find it easier to understand the condition with `== true` rather than `?? false`, but level of care low.

```suggestion
            if (queryStringManager?.ShouldObfuscate(queryString) == true)
```